source/slang/slang-ir-specialize-function-call.cpp

```diff
@@ -505,6 +508,11 @@ struct FunctionParameterSpecializationContext
             //
             ioInfo.key.vals.add(oldGlobalParam);
         }
+        else if( auto globalConstant = as<IRGlobalValueWithCode>(oldArg) )
```

Comment:
`IRGlobalValueWithCode` is not just `constant`.

Reply:
mm, perhaps that was poor wording. I meant it in the sense that the identity isn't variable which is what matters for specialization.